SCAutolib/models/gui.py

```diff
@@ -0,0 +1,296 @@
+             filename
+        ])
+
+        self.screenshot_num += 1
+        return filename
+
+
+class Mouse():
+    def __init__(self):
+        run(['modprobe', 'uinput'], check=True)
+        # initialize the uinput device
+        self.device = uinput.Device([
+            uinput.REL_X,
+            uinput.REL_Y,
+            uinput.BTN_LEFT,
+            uinput.BTN_MIDDLE,
+            uinput.BTN_RIGHT,
+            uinput.REL_WHEEL,
+        ])
+
+    def move(self, x, y):
+        '''Moves the mouse cursor to specified absolute coordinate.'''
+
+        # Go to upper left corner
+        self.device.emit(uinput.REL_X, -10000)
+        sleep(0.1)
+        self.device.emit(uinput.REL_Y, -10000)
+        sleep(0.1)
+
+        # Move to correct X coordinate
+        for i in range(x):
+            self.device.emit(uinput.REL_X, 1)
+            sleep(0.005)
+
+        # Move to correct Y coordinate
+        for i in range(y):
+            self.device.emit(uinput.REL_Y, 1)
+            sleep(0.005)
+
+    def click(self, button='left'):
+        '''
+        Clicks the any button of the mouse.
+
+        Button value can be 'left', 'right' or 'middle'.
+        '''
+
+        button_map = {
+            'left': uinput.BTN_LEFT,
+            'right': uinput.BTN_RIGHT,
+            'middle': uinput.BTN_MIDDLE,
+        }
+        uinput_button = button_map[button]
+
+        # press the button
+        self.device.emit(uinput_button, 1)
+        # wait a little
+        sleep(0.1)
+        # release the button
+        self.device.emit(uinput_button, 0)
+
+# Wrapper class for keyboard library
+class KB():
+    def __init__(self):
+        def kb_decorator(fn):
+            def wrapper(*args,**kwargs):
+                print('keyboard...')
+                fn(*args, **kwargs)
+                sleep(1)
+            return wrapper
+
+        # Workarounds for keyboard library
+        # keyboard.write types nothing if the delay is not set
+        self.write = kb_decorator(functools.partial(keyboard.write, delay=0.1))
+        self.send = kb_decorator(keyboard.send)
+        # For some reason the first keypress is never sent
+        # So this effectively does nothing
+        keyboard.send('enter')
+
+
+class GUI():
+    def __init__(self):
+        # Create the directory for screenshots
+        # TODO parametrize?
+        self.screenshot_directory = '/tmp/SC-tests'
+        os.makedirs(self.screenshot_directory, exist_ok=True)
+
+        self.mouse = Mouse()
+
+        self.kb = KB()
+
+
+    def __enter__(self):
+        self.screen = Screen(self.screenshot_directory)
+        # By restarting gdm, the system gets into defined state
+        run(['systemctl', 'restart', 'gdm'], check=True)
+        # Cannot screenshot before gdm starts displaying
+        # This would break the display
+        sleep(5)
+
+        return self
+
+
+    def __exit__(self, type, value, traceback):
+        # Gather the screenshots, generate report
+        pass
```

Comment:
It would be better to check for ant exception that happend inside the context manger and at least log them to the ouput